controller/src/server/functions.ts

```diff
@@ -14,18 +21,27 @@ import type * as types from '../types.js';
 
 dotenv.config();
 
-const hasura = process.env.HASURA ?? 'http://127.0.0.1:8080/v1/graphql';
-const client = new GraphQLClient(hasura, {
-  headers: {
-    'x-hasura-admin-secret': 'hasuraadminsecret',
-  },
-});
-const sdk = getSdk(client);
+let client: GraphQLClient;
+let sdk: { setRunAsQueued: (argument0: { run_id: string; }) => any,
+  AllSimulations: () => AllSimulationsQuery | PromiseLike<AllSimulationsQuery>,
+  allRunsAndSteps: () => any,
+  createSimulation: (argument0: { model_id: string; name: string; }) => any,
+  createStep: (argument0: { run_id: string; name: string; image: string; pipeline_step_number: number; }) => any,
+  createRun: (argument0: { simulation_id: string; dsl: string; name: string; }) => any,
+  setRunAsStarted: (argument0: { run_id: string; }) => any,
+  getRunDetails: (argument0: { run_id: string; }) => any,
+  getSimulationRunResults: (argument0: { simulation_id: string; run_id: string; }) => any,
+  setRunAsCancelled: (argument0: { run_id: string; }) => any,
+  setStepAsCancelled: (argument0: { step_id: any; }) => any,
+  setRunAsFailed: (argument0: { run_id: string; }) => any,
+  setRunAsEndedSuccess: (argument0: { run_id: string; }) => any
+};
 const uploadDirectory = 'uploaded_files/';
 
-export async function allSimulations():Promise<AllSimulationsQuery> {
-  // return JSON.stringify(await sdk.AllSimulations(), undefined, 2);
-  return await sdk.AllSimulations();
+export async function allSimulations():Promise<string> {
+// export async function allSimulations():Promise<AllSimulationsQuery> {
+  return JSON.stringify(await sdk.AllSimulations(), undefined, 2);
+  // return await sdk.AllSimulations();
```

Comment:
Why do we return a JSON string instead of simulation queries?

Reply:
This was done mainly to avoid declaring the types Simulation and Run just for this query. But I have changed this in the latest commit.